### DISCUSSION TITLE
Running tycho surefire tests with JustJ from target platform

### DISCUSSION THREAD

**Discussion by szarnekow** [eclipse-tycho/tycho#1263] [Q&A]
We bundle a specific JustJ version with our product and want to run the tests against that particular JustJ.

The target-platform-configuration allows to specify that the execution environment is available from the TP:

```xml
<plugin>
	<groupId>org.eclipse.tycho</groupId>
	<artifactId>target-platform-configuration</artifactId>
	<configuration>
		<executionEnvironment>org.eclipse.justj.openjdk.hotspot.jre.full-17</executionEnvironment>
		<resolveWithExecutionEnvironmentConstraints>true</resolveWithExecutionEnvironmentConstraints>
	</configuration>
</plugin>
```

For the surefire tests, we use `<useJDK>BREE</useJDK>` but that fails to match against the JRE from the TP. Is there a solution to that problem?

**Comment by phermsdorf** [comment]
We do run our tests in a docker container and have a toolchain.xml file in that conatiner that points to the bundled JustJ like this:
```
<toolchain>
     <type>jdk</type>
     <provides>
         <id>JavaSE-11</id>
         <version>11</version>
     </provides>
     <configuration>
        <jdkHome>/somepath/justj-java</jdkHome>
     </configuration>
  </toolchain>
```
But of course that does not work when running test directly or locally.

**Reply by szarnekow** [reply]
Indeed. We do have a docker file, too but it's managed by a different team and not versioned along with the code that is being built in that container. So the coupling there makes it quite hard to change the target platform and guarantee consistent test results.

What I'll likely be able to do is to materialize the product, create a toolchains-for-tests.xml pointing to the JRE in the product and launch the tests only afterwards. But that's really annoying because it complicates the build and does no longer allow a single maven command to run it.

**Comment by laeubi** [comment]
> For the surefire tests, we use `<useJDK>BREE</useJDK>` but that fails to match against the JRE from the TP.

I don't think this is supported, `useJDK` is refering to the toolchain as @phermsdorf has mentioned, I think it *might* be possible to use the one from the target but that will require some adjustments in Tycho if this is crucial to someones business and likes to speed up the development in that area a [sponsoring](https://github.com/sponsors/laeubi?frequency=one-time) would allow me to assign more time-slots particular issue.

**Reply by szarnekow** [reply]
> I think it might be possible to use the one from the target but that will require some adjustments in Tycho

Yes, I thought so too. I'll check what we can do about the sponsoring thingy.
I'm afraid there might be more surprises lurking in the grey area of the duality between the target platform's osgi-ee and the toolchains configured for plain maven.
